src/js/webrequest.js

```diff
@@ -263,23 +263,28 @@ function onHeadersReceived(details) {
   let tab_host = getHostForTab(tab_id);
   let response_host = window.extractHostFromURL(url);
 
+  let newHeaders = details.responseHeaders;
+  if (badger.isFlocOverwriteEnabled() && (details.type === 'main_frame' || details.type === 'sub_frame')) {
```

Comment:
I think we need to inject into top-level frames only, as more restrictive parent permissions policies should take precedence over policies declared by nested frames:
- https://developers.google.com/web/updates/2018/06/feature-policy#inheritancerules
- https://w3c.github.io/webappsec-permissions-policy/#examples (example 1)
- https://www.w3.org/TR/permissions-policy-1/#inherited-policies

Reply:
If only do this for top-level frames, we can then undo sending `newHeaders` in these new cases below, and instead inject the permissions policy header in just the "if this is a main frame and FLoC is supported and FLoC should be disabled" case and short circuit out of the rest of the listener (since "main_frame" requests should never get past the subsequent `isThirdPartyDomain` check).